bootstrap/ansible/playbook.yml

```diff
@@ -1,284 +1,491 @@
+          register: pgdata_dir_version
+
+        - name: Initialize PostgreSQL if not already initialized
+          command: /usr/pgsql-9.6/bin/postgresql96-setup initdb
+          when: not pgdata_dir_version.stat.exists
+
+        - name: Start and enable PostgreSQL service
+          service:
+            name: postgresql-9.6
+            state: started
+            enabled: yes
+
+        - name: Update pg_hba.conf to change host auth from ident to md5 (1/2)
+          postgresql_pg_hba:
+            dest: /var/lib/pgsql/9.6/data/pg_hba.conf
+            backup: yes
+            method: md5
+            contype: host
+            source: ::1
+
+        - name: Update pg_hba.conf to change host auth from ident to md5 (2/2)
+          postgresql_pg_hba:
+            dest: /var/lib/pgsql/9.6/data/pg_hba.conf
+            backup: yes
+            method: md5
+            contype: host
+            source: 127.0.0.1/32
+
+        - name: Update pg_hba.conf to set authentication for all to md5
+          postgresql_pg_hba:
+            dest: /var/lib/pgsql/9.6/data/pg_hba.conf
+            backup: yes
+            method: md5
+            contype: host
+            source: "0.0.0.0/0"
+
+        - name: Restart PostgreSQL service
+          service:
+            name: postgresql-9.6
+            state: restarted
+
+        # Create Django application log files.
+
+        - name: Create a parent directory for storing Django application logs
+          file:
+            path: "{{ log_path }}"
+            state: directory
+            mode: 0775
+            owner: "{{ djangooperator }}"
+            group: apache
+
+        - name: Create a log file for Django portal logs
+          copy:
+            content: ""
+            force: no
+            dest: "{{ log_path }}/{{ portal_log_file }}"
+            mode: 0660
+            owner: "{{ djangooperator }}"
+            group: apache
+
+        - name: Create a log file for Django API logs
+          copy:
+            content: ""
+            force: no
+            dest: "{{ log_path }}/{{ api_log_file }}"
+            mode: 0660
+            owner: "{{ djangooperator }}"
+            group: apache
+
+        # Prepare the Python virtual environment for the Django application.
+
+        - name: Create Python virtual environment directory
+          file:
+            path: "{{ git_prefix }}/venv"
+            state: directory
+            mode: 0755
+            owner: "{{ djangooperator }}"
+            group: apache
+
+        - name: Initialize Python virtual environment
+          shell: >
+              test -f {{ git_prefix }}/venv/bin/activate
+              || (python3.6 -m venv {{ git_prefix }}/venv
+              && echo created)
+          register: venv_register
+          become_user: "{{ djangooperator }}"
+          changed_when: venv_register.stdout == "created"
+
+        # Install Django application dependencies, so that psycopg2, which is
```

Comment:
We could make this common, since we do add requirements. It certainly isn't true to say that this "can be skipped always after the first time".